theme/views.py

```diff
@@ -120,6 +122,8 @@ def rating(request):
     obj, post_data = response
     url = add_cache_bypass(obj.get_absolute_url().split("#")[0])
     response = redirect(url + "#rating-%s" % obj.id)
+    resource_mode = post_data.get('resource-mode', 'view')
+    request.session['resource-mode'] = resource_mode
```

Comment:
It'd be best to delete a session variable after its usage is done by a different page request which is what we have been doing for other session variables. Otherwise, the session will only expire after browser closes according to our setting here: https://github.com/hydroshare/hydroshare/blob/baa3b03356b7d58e42f546881d610b74f87c13c0/hydroshare/settings.py. You may test to see whether this is an issue or not.